docs/sia_server_guide.md

```diff
@@ -0,0 +1,75 @@
+SIA Server is required for authenticating existing unmanaged services. For
+managed service inception (e.g., Manhattan), other more convenient
+automated mechanisms get the credentials safely to the service. The SIA
+server provides on-box support for securely getting an identity token
+for a running service that was previously registered in ZMS.
+
+The SIA server requires access to that service's [private key](reg_service_guide.md)
+to sign the token. The server runs as a protected local agent, and the services use Unix domain
+socket interface to request tokens. SIA communicates with ZTS to validate service details.
+
+## Installing SIA
+
+The AWS SIA source is part of the Athenz project and can be found in:
+```
+provider/aws/sia-ec2
+``` 
+Follow the readme for instructions on how to install it.
+
+## Private Key Installation
+
+The SIA Server needs to have access the service's private key so it can
+generate and sign a ServiceToken for the service. Then the service can
+use that ServiceToken to authenticate itself in Athenz and, if
+necessary, obtain a RoleToken to process requests against other Athenz
+protected services.
+
+## Key Update/New Service
+
+If a new service is updated on the host or the key has been updated on
+disk, the SIA service must be restarted for the changes to take effect
+because it caches the private key details during server startup.
+
+## Service Identity Setup Verification
+
+Once the private key has been installed on the SIA Server and the
+Service has been registered in ZMS with its public key, the
+administrator can verify that the setup is done correctly by executing
+the following command:
+
+```
+sudo /home/y/bin64/sia_setup_verify -d <domain-name> -s <service-name>
+```
+
+The command will verify that the private key is installed in the correct
+directory location and a service token will be generated successfully.
+Then it will verify that it can authenticate against ZTS server with
+that service token and retrieve the Service Identity details. It will
+also verify that the public key with the correct key identifier is
+registered in Athenz. Any issues will be reported with suggestions how
+to fix them.
+
+This utility only verifies your service configuration without connecting
+to SIA Server over its protocol. Once the configuration details have been
+verified, the administrator needs to make sure that the SIA Server is
+restarted and running.
```

Comment:
Removed the verification tool "sia_ntoken" which is part of "athens_tools" which isn't open source. Original text:
 restarted and running. You can restart sia server using yinst:

 ```
 yinst restart sia_server
 ```

 To verify that sia server is ready to issue service ntokens over its
 protocol, install [athens_tools](https://dist.corp.yahoo.com/by-package/athens_tools/)
 package on the host and execute:

 ```
 sudo /home/y/bin64/sia_ntoken -d <domain-name> -s <service-name>
 ```

 If SIA Server is running and configured correctly, the command will return
 a ServiceToken for the configured service.

Reply:
@havetisyan should we remove sia server completely?